apps/rig_inbound_gateway/test/event_buffer_test.exs

```diff
@@ -0,0 +1,82 @@
+defmodule RigInboundGatewayWeb.EventBufferTest do
+  use ExUnit.Case, async: true
+
+  alias RigCloudEvents.CloudEvent
+  alias RigInboundGatewayWeb.EventBuffer
+
+  defp new_event(id) do
+    {:ok, event} =
+      CloudEvent.parse("""
+        {"specversion": "0.2", "id": "#{id}", "type": "test-event"}
+      """)
+
+    event
+  end
+
+  test "Adding events leaves the capacity unchanged and overwrites old events as soon as the buffer is filled up to its capacity." do
+    # We create a new buffer that holds at most 2 events:
+    buffer = EventBuffer.new(2)
+    assert buffer |> EventBuffer.capacity() == 2
+    assert buffer |> EventBuffer.all_events() == []
+
+    # We add a first event:
+    first_event = new_event("first")
+    buffer = buffer |> EventBuffer.add_event(first_event)
+    # The event is in the buffer:
+    assert buffer |> EventBuffer.all_events() == [first_event]
+    # There is no more recent event yet:
+    {:ok, [events: events, last_event_id: last_event_id]} =
+      buffer |> EventBuffer.events_since("first")
+
+    assert events == []
+    assert last_event_id == "first"
+
+    # We add a second event to the buffer:
+    second_event = new_event("second")
+    buffer = buffer |> EventBuffer.add_event(second_event)
+    # Both events are now in the buffer:
+    assert buffer |> EventBuffer.all_events() == [first_event, second_event]
```

Comment:
imho, using a pipeline operator for a single function call makes the code a little bit harder to read.